plugins/modules/boot.py

```diff
@@ -404,7 +404,7 @@ def main():
                     if should is None:
                         continue
                     # unfold the return object for the idempotence check to work correctly
-                    has = existing.get(data_key)
+                    has = existing.get(result_key)
```

Comment:
While reviewing I noticed that it should be using `result_key` here, not `data_key` - this so far didn't make a difference since the values were equal, but now they're not :)